projects/common/src/utilities/formatters/date/display-date.pipe.ts

```diff
@@ -1,11 +1,14 @@
 import { Pipe, PipeTransform } from '@angular/core';
+import { isNil } from 'lodash-es';
+import { DateCoercer } from '../../coercers/date-coercer';
 import { DateFormatOptions, DateFormatter } from './date-formatter';
 
 @Pipe({
   name: 'htDisplayDate'
 })
 export class DisplayDatePipe implements PipeTransform {
-  public transform(value?: Date | number | null, options: DateFormatOptions = {}): string {
-    return value !== null ? new DateFormatter(options).format(value) : '-';
+  private readonly dateCoercer: DateCoercer = new DateCoercer();
+  public transform(value?: string | Date | number | null, options: DateFormatOptions = {}): string {
+    return isNil(value) ? '-' : new DateFormatter(options).format(this.dateCoercer.coerce(value));
```

Comment:
what if `this.dateCoercer.coerce(value)` returns undefined? Should we do this before the calling isNil  so that we can assign '-'?

Reply:
Should the DateFormatter just depend on dateCoercer?

Reply:
addressed